### PR TITLE
feat: dark theme

### PR DESCRIPTION
**Pull request by iCrawl**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/valentingavran/anglify/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by iCrawl** [comment]
Needs: #123

**Comment by iCrawl** [comment]
The addition of the material dark theme per spec needs us to change how we import themes to allow importing multiple themes and change them on the fly.

This means that themes are now exporting their css vars as mixins and all we have to do is `@include` the relevant mixin theme to where its needed.

File: apps/docs/src/styles.scss [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+  &.light {
+    @import '~highlight.js/scss/github.scss';
+  }
+
+  &.dark {
+    @import '~highlight.js/scss/github-dark.scss';
+    @include dark-theme;
+  }
+}
+
```
Comment (valentingavran) [inline apps/docs/src/styles.scss:19]: The imports in lines 11 & 15 should not be necessary:

```scss
@use '../../../libs/anglify/src/styles/index';
@use '../../../libs/anglify/src/styles/themes/light' as *;
@use '../../../libs/anglify/src/styles/themes/dark' as *;
@use '../../../libs/anglify/src/modules/table';
@use '../../../libs/anglify/src/styles/settings/typography.scss';

:root {
  @include light-theme;

  &.dark {
    @include dark-theme;
  }
}
```

**Review by valentingavran — CHANGES_REQUESTED** [review]
Please split up the PR into three commits (three PRs are not necessary):

1. refactoring commit, which introduces the breaking change that imports themes via mixins.
2. feature commit, which adds a dark theme
3. docs commit, where the common-tags library is added wherever needed in the documentation

Thank you and sorry for the inconvenience 🙏🏻

File: apps/docs/src/styles.scss (hunk 1 above)
Comment (iCrawl) [reply apps/docs/src/styles.scss:19]: Those were added by yourself, and they are necessary, I think. I did not modify those.

Comment (valentingavran) [reply apps/docs/src/styles.scss:19]: I am as dumb as a brick. I thought these were Anglify Style Imports 🥴

My bad. Thank you for questioning this

**Comment by iCrawl** [comment]
All done as requested.

**Review by valentingavran — APPROVED** [review]

**Pull request merged**